src/lib.rs

```diff
@@ -365,6 +373,17 @@ impl ExtendedSpendingKey {
         }
     }
 
+    pub fn write<W: Write>(&self, mut writer: W) -> io::Result<()> {
+        writer.write_u8(self.depth)?;
+        writer.write_all(&self.parent_fvk_tag.0)?;
+        writer.write_u32::<LittleEndian>(self.child_index.to_index())?;
+        writer.write_all(&self.chain_code.0)?;
+        writer.write_all(&self.xsk.to_bytes())?;
```

Comment:
Perhaps, a slight at least syntactic difference from the spec where `(xsk,dk)` are given together to `EncodeXSKParts`, rather than being serialized separately.